elide-contrib/elide-swagger/src/main/java/com/yahoo/elide/contrib/swagger/SwaggerBuilder.java

```diff
@@ -0,0 +1,708 @@
+            this.type = type;
+            lineage = new Stack<>();
+            name = dictionary.getJsonAliasFor(type);
+        }
+
+        /**
+         * @return Something like '/book/{bookId}/authors' or '/publisher'
+         */
+        public String getCollectionUrl() {
+            if (lineage.isEmpty()) {
+                return "/" + name;
+            } else {
+                return lineage.peek().getInstanceUrl() + "/" + name;
+            }
+        }
+
+        /**
+         * @return Something like '/book/{bookId}'
+         */
+        public String getInstanceUrl() {
+            String typeName = dictionary.getJsonAliasFor(type);
+            return getCollectionUrl() + "/{" + typeName + "Id}";
+        }
+
+        /**
+         * @return Something like '/book/{bookId}/relationships/authors'
+         */
+        public String getRelationshipUrl() {
+            if (lineage.isEmpty()) {
+                throw new IllegalStateException("Root collections don't have relationships");
+            }
+
+            PathMetaData prior = lineage.peek();
+            String baseUrl = prior.getInstanceUrl();
+
+            return baseUrl + "/relationships/" + name;
+        }
+
+        /**
+         * All Paths are 'tagged' in swagger with the root entity name in the path.
+         * This allows swaggerUI to group the paths by the root entities.
+         * @return the root entity name
+         */
+        private String getTag() {
+            if (lineage.isEmpty()) {
+                return name;
+            } else {
+                return lineage.get(0).getName();
+            }
+        }
+
+        /**
+         * @return the swagger PathParameter for this particular path segment.
+         */
+        private Parameter getPathParameter() {
+            String typeName = dictionary.getJsonAliasFor(type);
+
+            Parameter param = new PathParameter()
+                    .name(typeName + "Id")
+                    .description(typeName + " Identifier")
+                    .property(new StringProperty());
+
+            return param;
+        }
+
+        /**
+         * @return the Swagger 'Path' for a relationship URL (/books/{bookId}/relationships/author).
+         */
+        public Path getRelationshipPath() {
+            if (lineage.isEmpty()) {
+                throw new IllegalStateException("Root collections don't have relationships");
+            }
+
+            Path path = new Path();
+
+            /* The path parameter apply for all operations */
+            lineage.stream().forEach((item) -> {
+                path.addParameter(item.getPathParameter());
+            });
+
+            String typeName = dictionary.getJsonAliasFor(type);
+
+            Response okSingularResponse = new Response()
+                    .description("Successful response")
+                    .schema(new com.yahoo.elide.contrib.swagger.property.Datum(new com.yahoo.elide.contrib.swagger
```

Comment:
Yes.  Unfortunately swagger defines two nearly identical hierarchies - models & properties.  We can have a Datum which is a property or a Datum which is model - very confusing.